examples/fluids/qfunctions/channel.h

```diff
@@ -29,46 +30,70 @@ struct ChannelContext_ {
-    const CeedScalar X[], CeedInt Nf, CeedScalar q[], void *ctx) {
+CEED_QFUNCTION_HELPER State Exact_Channel(CeedInt dim, CeedScalar time,
+    const CeedScalar X[], CeedInt Nf, void *ctx) {
 
   const ChannelContext context = (ChannelContext)ctx;
-  const CeedScalar theta0 = context->theta0;
-  const CeedScalar P0     = context->P0;
-  const CeedScalar umax   = context->umax;
-  const CeedScalar center = context->center;
-  const CeedScalar H      = context->H;
-  const CeedScalar cv     = context->newtonian_ctx.cv;
-  const CeedScalar cp     = context->newtonian_ctx.cp;
-  const CeedScalar Rd     = cp - cv;
-  const CeedScalar mu     = context->newtonian_ctx.mu;
-  const CeedScalar k      = context->newtonian_ctx.k;
-
-  const CeedScalar y=X[1];
-
+  const CeedScalar theta0      = context->theta0;
+  const CeedScalar P0          = context->P0;
+  const CeedScalar umax        = context->umax;
+  const CeedScalar center      = context->center;
+  const CeedScalar H           = context->H;
+  NewtonianIdealGasContext gas = &context->newtonian_ctx;
+  const CeedScalar cp          = gas->cp;
+  const CeedScalar mu          = gas->mu;
+  const CeedScalar k           = gas->k;
+  // There is a gravity body force but it is excluded from
+  //   the potential energy due to periodicity.
+  gas->g[0] = 0.;
+  gas->g[1] = 0.;
+  gas->g[2] = 0.;
+
+  const CeedScalar y     = X[1];
   const CeedScalar Pr    = mu / (cp*k);
   const CeedScalar Ec    = (umax*umax) / (cp*theta0);
   const CeedScalar theta = theta0*(1 + (Pr*Ec/3)
                                    * (1 - Square(Square((y-center)/H))));
+  CeedScalar Y[5] = {0.};
+  Y[0] = P0;
+  Y[1] = umax*(1 - Square((y-center)/H));
+  Y[2] = 0.;
+  Y[3] = 0.;
+  Y[4] = theta;
+
+  return StateFromY(gas, Y, X);
+}
 
-  const CeedScalar p = P0;
+// *****************************************************************************
+// These QFunctions sets the initial condition
+// *****************************************************************************
+CEED_QFUNCTION(ICsChannel_Prim)(void *ctx, CeedInt Q,
+                                const CeedScalar *const *in, CeedScalar *const *out) {
+  // Inputs
+  const CeedScalar (*X)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[0];
 
-  const CeedScalar rho = p / (Rd*theta);
+  // Outputs
+  CeedScalar (*q0)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0];
 
-  q[0] = rho;
-  q[1] = rho * umax*(1 - Square((y-center)/H));
-  q[2] = 0;
-  q[3] = 0;
-  q[4] = rho * (cv*theta) + .5 * (q[1]*q[1] + q[2]*q[2] + q[3]*q[3]) / rho;
+  // Quadrature Point Loop
+  CeedPragmaSIMD
+  for (CeedInt i=0; i<Q; i++) {
+    const CeedScalar x[] = {X[0][i], X[1][i], X[2][i]};
+    State s = Exact_Channel(3, 0., x, 5, ctx);
+    CeedScalar q[5] = {0.};
+    q[0] = s.Y.pressure;
+    for (CeedInt j=0; j<3; j++)
+      q[j+1] = s.Y.velocity[j];
+    q[4] = s.Y.temperature;
 
+    for (CeedInt j=0; j<5; j++)
+      q0[j][i] = q[j];
+  } // End of Quadrature Point Loop
```

Comment:
Probably just condense into a single loop?
```suggestion
    q0[0][i] = s.Y.pressure;
    for (CeedInt j=0; j<3; j++)
      q0[j+1][i] = s.Y.velocity[j];
    q0[4][i] = s.Y.temperature;
  } // End of Quadrature Point Loop
```